src/v/raft/group_manager.cc

```diff
@@ -203,4 +223,19 @@ void group_manager::setup_metrics() {
         sm::description("Number of raft groups"))});
 }
 
+ss::future<> group_manager::flush_groups() {
+    auto u = co_await _groups_mutex.get_units();
+
+    co_await ss::max_concurrent_for_each(
+      _groups.begin(),
+      _groups.end(),
+      32,
+      [this](const ss::lw_shared_ptr<consensus>& raft) {
+          if (_configuration.replica_max_not_flushed_bytes()) {
+              return raft->maybe_flush_log(
+                _configuration.replica_max_not_flushed_bytes().value());
```

Comment:
So this will leave a small not-flushed tail at the end of each log, leading to a small persistent difference between dirty and flushed offset, right? I am not familiar with performance characteristics of high-rate flushes, but perhaps we can provide an entirely timer-based guarantee by limiting concurrency/rate of flushes somehow? IMO it would be a simpler guarantee to understand.

Reply:
I have a proposal, i will add time based check in a separate follow up PR, wdyt ?

Reply:
sounds good!